tests/integration/sync_test.go

```diff
@@ -0,0 +1,106 @@
+//go:build integration
+
+package integration
+
+import (
+	"context"
+	"os"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	"github.com/kong/deck/cmd"
+	"github.com/kong/deck/dump"
+	"github.com/kong/deck/utils"
+	"github.com/kong/go-kong/kong"
+)
+
+var deckCmd = cmd.RootCmdOnlyForDocsAndTest
+
+func getKongAddress() string {
+	address := os.Getenv("DECK_KONG_ADDR")
+	if address != "" {
+		return address
+	}
+	return "http://localhost:8001"
+}
+
+// cleanUpEnv removes all existing entities from Kong.
+func cleanUpEnv(t *testing.T) error {
+	deckCmd.SetArgs([]string{"reset", "--force"})
+	if err := deckCmd.Execute(); err != nil {
+		t.Errorf(err.Error())
+	}
+	return nil
+}
+
+func getTestClient() (*kong.Client, error) {
+	return utils.GetKongClient(utils.KongClientConfig{
+		Address: getKongAddress(),
+	})
+}
+
+func testKongState(
+	t *testing.T,
+	client *kong.Client,
+	expectedServices []*kong.Service,
+) {
+	ctx := context.Background()
```

Comment:
Use a timeout. In this codebase, it will be rare (if there is one) to do an operation without a timeout.